ray_data_eval/simulator/policies.py

```diff
@@ -124,3 +124,71 @@ def on_task_state_change(self, task: TaskSpec, task_state: TaskState):
             self._buffer_diff[self._current_tick + task.duration] += (
                 task.output_size - task.input_size
             )
+
+
+class EqualInputOutputRateSchedulingPolicy(SchedulingPolicy):
+    """
+    A policy that schedules consumer to be the same rate as the producer
+    """
+
+    def __init__(self, problem: SchedulingProblem):
+        super().__init__(problem)
+        self.buffer_size_limit = problem.buffer_size_limit
+        self._current_tick = 0
+        self._cumulative_output_size = 0
+        producer_rate = problem.producer_output_size[0] / problem.producer_time[0]
+        consumer_rate = problem.consumer_input_size[0] / problem.consumer_time[0]
+        self.producer_consumer_ratio = producer_rate / consumer_rate
+
+    def __repr__(self):
+        return "EqualInputOutputRateSchedulingPolicy"
+
+    def is_producer_task(self, env: ExecutionEnvironment, tid: int):
+        return env.task_specs[tid].output_size > 0
+
+    def tick(self, env: ExecutionEnvironment):
+        super().tick(env)
+        num_producer = 0
+        num_consumer = 0
+
+        for tid, task_state in env.task_states.items():
```

Comment:
move this into a helper function: `num_producers, num_consumers = _count_scheduled_tasks(env.task_states)`